lib/ansible/modules/network/junos/junos_netconf.py

```diff
@@ -139,7 +139,7 @@ def map_params_to_obj(module):
     for key, value in iteritems(obj):
         # validate the param value (if validator func exists)
         validator = globals().get('validate_%s' % key)
-        if all((value, validator)):
+        if callable(validator):
```

Comment:
For below playbook:
```
    - name: Configure Netconf
      junos_netconf:
        state: present
        netconf_port:
```
The ouptut is
```
changed: [vsrx01] => {
    "changed": true, 
    "commands": [
        "set system services netconf ssh port None"
    ], 
    "invocation": {
        "module_args": {
            "host": null, 
            "netconf_port": null, 
            "password": null, 
            "port": null, 
            "provider": {
                "host": "vsrx01", 
                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "port": null, 
                "ssh_keyfile": null, 
                "timeout": null, 
                "transport": null, 
                "username": "ansible"
            }, 
            "ssh_keyfile": null, 
            "state": "present", 
            "timeout": 10, 
            "transport": null, 
            "username": null
        }
    }
}
```
So the port value is set to `None` even if it has a default value.

I agree with your point that `globals()` way in not as much readable, but it is extensible in a sense adding new validation requires adding only validation function without change to existing code.